devito/interfaces.py

```diff
@@ -151,7 +152,10 @@ def _indices(cls, **kwargs):
                 raise ValueError("Unknown symbol dimensions or shape")
             _indices = [x, y, z]
             shape = kwargs.get('shape')
-            dimensions = _indices[:len(shape)]
+            if len(shape) <= 3:
+                dimensions = _indices[:len(shape)]
+            else:
+                dimensions = [symbols("x%d" % i) for i in range(1, len(shape) + 1)]
```

Comment:
The earlier implementation was limited to 3 spatial dimensions and 1 time dimension. Anything bigger than that would fail.
This makes that more generic. For cases where we are using up to 3 spatial dimensions, we use the indices `x`, `y` and `z` - the same as before. When we have a `DenseData` with more than 3 spatial dimensions, these indices stop making sense as default. So the default in that case (let's say I have a 7 dimensional tensor as a `DenseData` object) is to have the indices like `x1`-`x7`.